cmd/snap-mgmt/snap-mgmt-selinux.sh.in

```diff
@@ -0,0 +1,113 @@
+#!/bin/bash
+
+set -e
+
+SNAP_MOUNT_DIR="@SNAP_MOUNT_DIR@"
+
+show_help() {
+    exec cat <<'EOF'
+Usage: snap-mgmt-selinux.sh [OPTIONS]
+
+A helper script to manage SELinux contexts used by snapd
+
+Arguments:
+  --snap-mount-dir=<path>                   Provide a path to be used as $SNAP_MOUNT_DIR
+  --patch-selinux-mount-context=<context>   Add SELinux context to mount units
+  --remove-selinux-mount-context=<context>  Remove SELinux context from mount units
+EOF
+}
+
+SNAP_UNIT_PREFIX="$(systemd-escape -p ${SNAP_MOUNT_DIR})"
+
+patch_selinux_mount_context() {
+    if ! command -v selinuxenabled > /dev/null; then
+        return
+    fi
+    if ! selinuxenabled; then
+        # the tools are there, but SELinux is not enabled
+        return
+    fi
+
+    selinux_mount_context="$1"
+    remove="$2"
+    if ! echo "$selinux_mount_context" | grep -qE '[a-zA-Z0-9_]+(:[a-zA-Z0-9_]+){2,3}'; then
+        echo "invalid mount context '$selinux_mount_context'"
+        exit 1
+    fi
+    context_opt="context=$selinux_mount_context"
+
+    units=$(systemctl list-unit-files --no-legend --full | grep -vF snap.mount.service || true)
```

Comment:
`systemctl list-unit-files` supports optional pattern argument, so this could become `... systemctl list-unit-files --no-legend --full snap.mount.service` with no grep needed, or am is there more to it?